stdlib/public/RemoteInspection/TypeLowering.cpp

```diff
@@ -2000,7 +2003,7 @@ class EnumTypeInfoBuilder {
     // same as cases with no payload, and generic cases are always considered
     // non-empty.
     unsigned EffectiveNoPayloadCases = NonPayloadCases + NonGenericEmptyPayloadCases;
-    unsigned EffectivePayloadCases = GenericPayloadCases + NonGenericNonEmptyPayloadCases;
+    unsigned EffectivePayloadCases = IndirectPayloadCases + GenericPayloadCases + NonGenericNonEmptyPayloadCases;
```

Comment:
Indirect cases do carry a payload, so we need to consider them payload cases for layout purposes.